src/pyfmi/fmi.pyx

```diff
@@ -1359,15 +1440,27 @@ cdef class FMUModelBase(ModelBase):
         self._modelname = decode(FMIL.fmi1_import_get_model_name(self._fmu))
         self._nEventIndicators = FMIL.fmi1_import_get_number_of_event_indicators(self._fmu)
         self._nContinuousStates = FMIL.fmi1_import_get_number_of_continuous_states(self._fmu)
-        fmu_log_name = encode((self._modelid + "_log.txt") if log_file_name=="" else log_file_name)
-        self._fmu_log_name = <char*>FMIL.malloc((FMIL.strlen(fmu_log_name)+1)*sizeof(char))
-        FMIL.strcpy(self._fmu_log_name, fmu_log_name)
 
-        #Create the log file
-        with open(self._fmu_log_name,'w') as file:
+        if not isinstance(log_file_name, str):
+            self._set_log_stream(log_file_name)
             for i in range(len(self._log)):
-                file.write("FMIL: module = %s, log level = %d: %s\n"%(self._log[i][0], self._log[i][1], self._log[i][2]))
-            self._log = []
+                try:
+                    self._log_stream.write("FMIL: module = %s, log level = %d: %s\n"%(self._log[i][0], self._log[i][1], self._log[i][2]))
+                except:
+                    if hasattr(self._log_stream, 'closed') and self._log_stream.closed:
+                        logging.warning("Unable to log to closed stream.")
+                    logging.warning("Unable to log to stream.")
```

Comment:
Should this not be in an else clause? Otherwise we get two warnings if the log has a closed attribute

Reply:
Good catch, fixed as well

Reply:
Hmm, does not seem that the else is added?

Reply:
Ah sorry there were two places in the code with this exact code, added in the second place as well:
https://github.com/modelon-community/PyFMI/commit/5103e9188ad872431f37e08004298ccde48f17ad